python/kserve/kserve/model.py

```diff
@@ -317,9 +317,20 @@ async def _http_predict(
         if isinstance(payload, InferRequest):
             payload = payload.to_rest()
         data = orjson.dumps(payload)
-        response = await self._http_client.post(
-            predict_url, timeout=self.timeout, headers=predict_headers, content=data
-        )
+
+        try:
+            response = await self._http_client.post(
+                predict_url, timeout=self.timeout, headers=predict_headers, content=data
+            )
+        except Exception as exc:
+            logger.error(
+                f"Could not send a request to predictor at url {predict_url} "
+                f"with headers {predict_headers} "
+                f"and with data {data} "
```

Comment:
What if there's sensitive data?

Reply:
Hi Yuan, I had this debate internally, thanks for raising it. The companies that need to protect their logs, do not rely on the libraries they use not to spew any data. They capture logs, scrub and obfuscate them, or hide them all together.

I think in this regard Kserve's priority should be providing a good developer experience. Its concern on privacy should not go beyond ensuring that logs are written only to `stdout` or `stderr` and nowehere else. This way the users who are interested in processing the logs, can do so safely.

What do you think?